src/Table/Table.js

```diff
@@ -44,17 +61,67 @@ const Table = React.forwardRef(({ headers, tableData, className, tableBodyClassN
         tableCheckboxClassName
     );
 
+    const useHookWithRefCallback = () => {
+        const newRef = useRef(null);
+        const setRef = useCallback(node => {
+            if (node && keyboardNavigation !== 'none') {
+                gridManager.current.attachTo({ gridNode: node, onFocusCell, onToggleEditMode });
+            }
+            newRef.current = node;
+        }, []);
+
+        return setRef;
+    };
+
+    const captionId = shortid.generate();
+    const gridManager = useRef(new GridManager());
+    const tableRef = ref || useHookWithRefCallback();
+    const [instructionsText, setInstructionsText] = useState('');
+
+    const onToggleEditMode = (enable) => {
+        setInstructionsText(enable ? localizedText.editModeDisable : '');
+    };
+
+    const onFocusCell = (cell, event) => {
+        const { row, col } = cell;
+        const key = event.which || event.keyCode;
+        const navigatedHorizontally = (key === keycode.codes.left || key === keycode.codes.right) && col > 0;
+        const navigatedVertically = (key === keycode.codes.up || key === keycode.codes.down) && row > 0;
+        if (gridManager.current?.editMode) {
+            setInstructionsText(localizedText.editModeDisable);
+        } else {
+            let newInstructionsText = '';
+            if (navigatedVertically) {
+                newInstructionsText += `${localizedText.row} ${row} `;
+            } else if (navigatedHorizontally) {
+                newInstructionsText += `${localizedText.column} ${col} ${headers[col]} `;
+            } else {
+                newInstructionsText += `${localizedText.arrowKeys} `;
+            }
+            if (gridManager.current?.isEditableCell(cell)) {
+                newInstructionsText += localizedText.editModeEnable;
+            }
+            setInstructionsText(newInstructionsText);
+        }
+    };
+
     let checkboxHeader;
-    let displayHeaders = headers;
+    let displayHeaders = [...headers];
 
     if (richTable) {
         checkboxHeader = <th className={tableCheckboxClasses}>{headers[0]}</th>;
-        displayHeaders = headers.splice(1, headers.length);
+        displayHeaders = displayHeaders.splice(1, headers.length);
     }
 
     return (
-        <table {...props} className={tableClasses}
-            ref={ref}>
+        <table {...props} aria-describedBy={captionId}
+            className={tableClasses}
+            ref={tableRef}
+            role={keyboardNavigation ? 'grid' : 'table'}>
+            <caption aria-live='polite' className='fd-table__instructions'
```

Comment:
Is `__instructions` intentional? In fd-styles the class for `<caption>` appears as `fd-table__caption`. Currently the text is visible when traversing through the cells.

Reply:
I decided to change it to `__caption` in the end, but didn't put that change here yet . Thanks for catching that.